api/src/main/java/edu/wpi/first/shuffleboard/api/sources/SourceType.java

```diff
@@ -61,4 +64,11 @@ public String removeProtocol(String text) {
     return sourceSupplier.apply(removeProtocol(uri));
   }
 
+  /**
+   * Gets a list of the IDs of all available sources of this type.
+   */
+  public ObservableList<String> getAvailableSourceIds() {
```

Comment:
1. What's a source ID? sources don't have IDs, they have URIs
2. this should be more explicit what "available" means (all currently known sources) and that this is only some subset of all sources and may not be implemented for every type